### PR TITLE
fix: screenshot windows

### PR DESCRIPTION
**Pull request by michaelneale**
Adds a tool to enumerate open windows, which allows screen capture of a specific window based on vague description. Whole screen capture I found didn't work well with multiple monitors and not at all if I wanted something specific. 

Screenshot 2024-12-09 at 5 39 00 PM
Screenshot 2024-12-09 at 5 39 10 PM

asking goose to match a design and iterate: 

image

### CODE REVIEW

**Review by Kvadratni — APPROVED** [review]
This is great. One thing - what if I have several windows of the same app?

**Comment by michaelneale** [comment]
@Kvadratni I expect it can look at all of them if needed

File: ui/desktop/src/main.ts [hunk 1]
```diff
@@ -315,10 +315,26 @@ app.whenReady().then(async () => {
 
   Menu.setApplicationMenu(menu);
 
-  app.on('activate', () => {
+  app.on('activate', async () => {
     if (BrowserWindow.getAllWindows().length === 0) {
       createChat(app);
     }
+    const sources = await electron.desktopCapturer.getSources({ types: ['screen'] })
```
Comment (michaelneale) [inline ui/desktop/src/main.ts]: this is to try to trigger a request for permissions early

Comment (alexhancock) [reply ui/desktop/src/main.ts]: Does it req the permissions right when the app launches?

Users will be more primed to accept if it happens when they try to do something that needs the access.

Comment (michaelneale) [reply ui/desktop/src/main.ts]: hrm - will at both in theory. I need to play around with it more to be sure. I can get rid of this so that it asks later

**Pull request merged**